bundle/src/main/java/com/adobe/acs/commons/properties/impl/PropertyAggregatorServiceImpl.java

```diff
@@ -0,0 +1,273 @@
+        }
+    }
+
+    private boolean notInMap(final Map<String, Object> map, final String propertyName) {
+        return !map.containsKey(prefix(PAGE_PROP_PREFIX, propertyName))
+                && !map.containsKey(prefix(INHERITED_PAGE_PROP_PREFIX, propertyName));
+    }
+
+    private boolean isNotExcluded(final String propertyName) {
+        for (Pattern pattern : exclusionList) {
+            if (pattern.matcher(propertyName).matches()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private String prefix(final String prefix, final String propertyName) {
+        return prefix + "." + propertyName;
+    }
+
+    private boolean isAllowedType(Object object) {
+        return String.class.equals(object.getClass()) || Long.class.equals(object.getClass());
+    }
+
+    /**
+     * Creates a map of configuration values based on the array and delimiter passed with the option
+     * to swap the key and value from the split string.
+     *
+     * @param configs The String array containing the delimited config values
+     * @return The map of config values
+     */
+    private Map<String, String> getMapFromArray(final String[] configs) {
+        Map<String, String> splitConfigs = new HashMap<>();
+        for (String config : configs) {
+            if (config.contains("|")) {
+                String[] parts = config.split(Pattern.quote("|"));
+                if (parts.length == 2) {
+                    splitConfigs.put(parts[0], parts[1]);
+                }
+            }
+        }
+        return splitConfigs;
+    }
+
+    @Activate
+    protected void activate(Config config) {
+        List<Pattern> excludeList = new ArrayList<>();
+        for (String exclude : config.exclude_list()) {
+            excludeList.add(Pattern.compile(exclude));
+        }
+        this.exclusionList = excludeList;
+        this.additionalData = getMapFromArray(config.additional_data());
+    }
+
+    @ObjectClassDefinition(
+            name = "Property Aggregator Service Configuration"
+    )
+    @interface Config {
+
+        /**
+         * The list of patterns or strings to exclude from the property aggregation.
+         *
+         * @return The list of exclusions
+         */
+        @AttributeDefinition(
+                name = "Exclude List",
+                description = "List of properties to exclude, accepts regex.",
+                type = AttributeType.STRING
+        )
+        String[] exclude_list() default {"cq:(.*)"};
+
+        /**
+         * The list of additional data sources to the map.
+         *
+         * @return The list of additional data
+         */
+        @AttributeDefinition(
+                name = "Additional Data",
+                description = "List of additional data sources. Each class provided should map to a"
+                        + " Sling Model that adapts from the content resource of a page. "
+                        + "Format is `prefix|className`",
```

Comment:
What is "prefix" here?

Reply:
The tool works with the expression `{{prefix.property}}` - i.e. `{{page_properties.pageTitle}}`. There needs to be a prefix defined for the custom classes added so that the code can choose which model is tied to the data that should be replaced.